kiwi/bootloader/config/grub2.py

```diff
@@ -641,7 +641,7 @@ def _setup_default_grub(self):
             'GRUB_TERMINAL': '"{0}"'.format(self.terminal)
         }
         grub_final_cmdline = re.sub(
-            r'root=.* |root=.*$', '', self.cmdline
+            r'root=[^\s]+', '', self.cmdline
```

Comment:
I think the expression should include the ' ' (space) we expect in front of the `root` attribute. Otherwise if there is ever another attribute that ends with `root` it will also get substituted. For example

```
>>> a = 'bootroot=/dev/foo root=/dev/bar hi'
>>> re.sub(r'root=[^\s]+', '', a)
'boot  hi'
>>> re.sub(r' root=[^\s]+', '', a)
'bootroot=/dev/foo hi'
```

Reply:
yes good point, thanks much :+1:  I'll adapt

Reply:
done